### ISSUE TITLE
Bug: `idempotentLambdaHandler` breaks when used with other decorators

### ISSUE THREAD

**Issue by dreamorosi** [aws-powertools/powertools-lambda-typescript#1481]
### Expected Behaviour

As a customer I should be able to use multiple class method decorators in conjunction with Idempotency's `idempotentLambdaHandler` decorator. The decorator should do its job and either allow other decorators to run or bail cleanly.

### Current Behaviour

When using multiple decorators (i.e. `idempotentLambdaHandler` & `logger.injectLambdaContext`) with `idempotentLambdaHandler` being the first one, the function throws an error like this:

> {"errorType":"TypeError","errorMessage":"Cannot read properties of undefined (reading 'invokedFunctionArn')","trace":["TypeError: Cannot read properties of undefined (reading 'invokedFunctionArn')","    at Logger.addContext (/var/task/index.js:3584:35)","    at Logger.injectLambdaContextBefore (/var/task/index.js:3662:13)","    at descriptor.value [as functionToMakeIdempotent] (/var/task/index.js:3643:16)","    at IdempotencyHandler.getFunctionResult (/var/task/index.js:1203:31)","    at IdempotencyHandler.processIdempotency (/var/task/index.js:1243:21)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async IdempotencyHandler.handle (/var/task/index.js:1223:20)"]}

### Code snippet

```typescript
import type { Context } from "aws-lambda";
import type { LambdaInterface } from "@aws-lambda-powertools/commons";
import { Logger } from "@aws-lambda-powertools/logger";
import { idempotentLambdaHandler } from "@aws-lambda-powertools/idempotency";
import { DynamoDBPersistenceLayer } from "@aws-lambda-powertools/idempotency/dynamodb";
import { IdempotencyConfig } from "@aws-lambda-powertools/idempotency";

type EventRecord = {
  foo: string;
};

const logger = new Logger();

const dynamoDBPersistenceLayer = new DynamoDBPersistenceLayer({
  tableName:
    process.env.IDEMPOTENCY_TABLE_NAME || "lambda-powertools-playground-table",
});
const idempotencyConfig = new IdempotencyConfig({});

class Lambda implements LambdaInterface {
  @idempotentLambdaHandler({
    persistenceStore: dynamoDBPersistenceLayer,
  })
  @logger.injectLambdaContext({ logEvent: true })
  public async handler(event: EventRecord, _context: Context): Promise<string> {
    return "Processing done: " + event.foo;
  }
}

const defaultLambda = new Lambda();
export const handler = defaultLambda.handler.bind(defaultLambda);
```

### Steps to Reproduce

1. Deploy the function above
2. Run it
3. Observe the error

### Powertools for AWS Lambda (TypeScript) version

latest

### AWS Lambda function runtime

18.x

### Packaging format used

npm

**Comment by dreamorosi** [comment]
Need to look into it further, but the stack trace shows that the Logger decorator is not able to read info from the Lambda context object. This suggests that the `idempotentLambdaHandler` decorator is somehow mutating the decorated method preventing subsequent decorators to read the context.

Swapping the order of decorators (i.e. `idempotentLambdaHandler` last) works normally.

**Comment by dreamorosi** [comment]
Since we have closed #1304 and #1306 and decided to not implement decorators for this utility, we are closing this issue for the time being.
